usage/rest-client/src/main/java/org/apache/brooklyn/rest/client/BrooklynApiFactory.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.brooklyn.rest.client;
+
+import java.net.URL;
+
+/**
+ * Creates {@link BrooklynApi} instances.
+ */
+public interface BrooklynApiFactory {
```

Comment:
why would i want to create two `BrooklynApi` instances sharing a pool?  feels like that use case isn't compelling enough to justify having a factory.  if i wanted pool-sharing behaviour i'd use the same `BrooklynApi` instance.  (if that's not allowed for thread reasons then i'd rather have a wrapper which does make it thread safe, as opposed to having to keep a handle on a factory instance.)